src/main/scala/subsystem/BusTopology.scala

```diff
@@ -0,0 +1,27 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.subsystem
+
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.tilelink._
+
+trait HasHierarchicalBusTopology { this: BaseSubsystem =>
```

Comment:
Nice!

Reply:
You can imagine mixing in a different set of inter-bus connections while using the same-named attachment points for all the devices...

Reply:
Yeah I actually already do this downstream but I had to duplicate more code than I wanted to :smile: